### PR TITLE
Handle zoom while camera change

### LINKED ISSUE #483: Switching from perspective projection to orthogonal projection zooms out a lot
Switching from perspective projection to orthogonal projection zooms out a lot. We should try to improve this.

https://github.com/user-attachments/assets/d0ac649d-6686-4298-a98a-aa188e3660ad

### PR DESCRIPTION
**Pull request by arjxn-py**
https://github.com/user-attachments/assets/edb44fbc-1fe9-401e-9156-63a3aa6a10c3

Shall fix #483

### CODE REVIEW

File: packages/base/src/3dview/mainview.tsx [hunk 1]
```diff
@@ -1323,12 +1324,17 @@ export class MainView extends React.Component<IProps, IStates> {
       const width = this.divRef.current?.clientWidth || 0;
       const height = this.divRef.current?.clientHeight || 0;
 
+      const distance = position.distanceTo(target);
+      const zoomFactor = 1000 / distance;
```
Comment (martinRenou) [inline packages/base/src/3dview/mainview.tsx:1328]: Do you know why we need this magic `1000` number?

Comment (arjxn-py) [reply packages/base/src/3dview/mainview.tsx:1328]: This is kind of an arbitrary number after some hit and trials, I noticed that more you zoom the object in `PerspectiveCamera` more further it goes on `OrthographicCamera`

Comment (arjxn-py) [reply packages/base/src/3dview/mainview.tsx:1328]: So it meant that the distance was inversely proportional to the zooming in that i needed to do

Comment (martinRenou) [reply packages/base/src/3dview/mainview.tsx:1328]: Sounds good! I tested on my side and this seems to work well. We have other magic numbers in JupyterCAD

**Pull request merged**